control/server.py

```diff
@@ -197,6 +198,18 @@ def serve(self):
             log_level = log_level.upper()
             log_req = pb2.set_spdk_nvmf_logs_req(log_level=log_level, print_level=log_level)
             self.gateway_rpc.set_spdk_nvmf_logs(log_req)
+        
+        self._register_service_map()
+
+    def _register_service_map(self):
+        # show gateway in "ceph status" output
+        conn = self.omap_state.conn
+        metadata = {
+            "id": re.sub(r'^client\.nvmeof\.', '', self.name),
```

Comment:
made this change because having `id` of "client.nvmeof.mypool.smithi049.hhvikx" was causing these warnings:
```
[WRN] CEPHADM_STRAY_DAEMON: 1 stray daemon(s) not managed by cephadm
    stray daemon nvmeof.client.nvmeof.mypool.smithi049.hhvikx on host smithi049 not managed by cephadm
```
so changed `id` to be "mypool.smithi049.hhvikx" instead. This fixes the cephadm warning.

Reply:
This is a huge overkill. No need to use re here. You can use:

self.name.removeprefix("client.nvmeof.")

instead.

Reply:
Thanks! I default to choosing regex for all kinds of string manipulations but its definitely overkill in this case :)